Clean dexcom history

channemann opened the pull request:
Functional for dexcom glucose histories; specifically, handles the different entry keys ('glucose' instead of 'sgv' or 'amount' and 'display_time' instead of 'date') as well as throws out low values that indicate errors.

channemann committed:
Added 'glucose' as entry

channemann committed:
Date parse

Dex and MM timestamps have different keys in JSON files; changed to
look for either

channemann committed:
Sorting error

Sorting procedure tried to sort on ‘time’ entry; added option for
‘display_time’ entries

channemann committed:
Else:

channemann committed:
Min glucose value for dex

Changed the minimum glucose value from > 0 to > 35 for the dexcom; low
values are for when there is an error or the sensor is warming up

channemann committed:
Merge pull request #1 from channemann/clean-dexcom-history

Clean dexcom history

channemann committed:
Added test for dex

channemann committed:
Merge branch 'master' into clean-dexcom-history

channemann committed:
Merge branch 'master' into dev

channemann committed:
Merge branch 'dev' into clean-dexcom-history

channemann commented:
Added unit test for dex. I wasn't entirely certain how to split it until multiple unit tests, so it still runs as the same one with two sections. Additionally, renamed the original test fixutre to iter_glucose_mm.json to differentiate.

loudnate commented on tests/glucose_tests.py:
```diff
@@ -11,7 +11,9 @@ def get_file_at_path(path):
 
 class CleanTestCase(unittest.TestCase):
     def test_clean_values(self):
```
Let's make two separate functions instead: `test_clean_values_mm` and `test_clean_values_dex`

loudnate commented on openapscontrib/mmglucosetools/glucose.py:
```diff
@@ -17,5 +19,9 @@ def clean(glucose_entries):
     """
     cleaned = filter(_filter_entry, glucose_entries)
-
-    return sorted(cleaned, key=lambda x: x['date'], reverse=True)
```
You can actually turn this into one line:

``` python
return sorted(cleaned, key=lambda x: x.get('date', x.get('display_time')), reverse=True)
```

channemann committed:
Implemented Nate's suggestions

Split testing into two functions (one for mm, one for sex); combined
return line if/elif statements into single lambda function

channemann commented:
Changes implemented, passed both tests on my Pi:

```
pi@raspberrypi ~/openaps-mmglucosetools $ sudo python setup.py test
running test
running egg_info
writing requirements to openapscontrib.mmglucosetools.egg-info/requires.txt
writing openapscontrib.mmglucosetools.egg-info/PKG-INFO
writing namespace_packages to openapscontrib.mmglucosetools.egg-info/namespace_packages.txt
writing top-level names to openapscontrib.mmglucosetools.egg-info/top_level.txt
writing dependency_links to openapscontrib.mmglucosetools.egg-info/dependency_links.txt
writing requirements to openapscontrib.mmglucosetools.egg-info/requires.txt
writing openapscontrib.mmglucosetools.egg-info/PKG-INFO
writing namespace_packages to openapscontrib.mmglucosetools.egg-info/namespace_packages.txt
writing top-level names to openapscontrib.mmglucosetools.egg-info/top_level.txt
writing dependency_links to openapscontrib.mmglucosetools.egg-info/dependency_links.txt
reading manifest file 'openapscontrib.mmglucosetools.egg-info/SOURCES.txt'
writing manifest file 'openapscontrib.mmglucosetools.egg-info/SOURCES.txt'
running build_ext
test_clean_values_dex (tests.glucose_tests.CleanTestCase) ... ok
test_clean_values_mm (tests.glucose_tests.CleanTestCase) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.016s

OK
```

loudnate commented:
Nice work!
